### PR TITLE
Fix warnings, leaks and null pointer access

### PR DESCRIPTION
**Pull request by mtbkeith**
There are several common issues fixed such as a possible null pointer to be used, division by zero, and memory leaks to happen when a new'd object is not deleted. A smaller nit was that parameters are not marked as unused and therefore many compiler messages get output due to this condition.

### CODE REVIEW

**Comment by liversedge** [comment]
NICE!

Its dull work, but really appreciate this, thank you.

**Comment by liversedge** [comment]
Did you perform an analysis with Coverity?
If so, any hints on how we can correct the config here in github?

**Comment by mtbkeith** [comment]
Hi @liversedge The analysis was performed with XCode using 'Analyze'. I was surprised at how it rivalled some of the other commercial tools I've used, however not a true test as I still haven't been able to d/l Coverity. It still shows "Please contact project owner/administrator to submit a build for the project."

File: src/Charts/AllPlot.cpp [hunk 1]
```diff
@@ -4968,6 +4972,9 @@ AllPlot::setDataFromPlots(QList<AllPlot *> plots)
                 }
         }
 
+        if (ourCurve) delete ourCurve;
+        if (ourICurve) delete ourICurve;
+
```
Comment (mtbkeith) [inline src/Charts/AllPlot.cpp]: deleting these curves is a bug; see https://github.com/GoldenCheetah/GoldenCheetah/pull/2438 for backing this change out.

File: src/Cloud/CloudService.cpp [hunk 2]
```diff
@@ -1344,6 +1344,7 @@ CloudServiceSyncDialog::downloadNext()
             QByteArray *data = new QByteArray; // gets deleted when read completes
             store->readFile(data, curr->text(1), curr->text(6));
             QApplication::processEvents();
+            delete data;
```
Comment (mtbkeith) [inline src/Cloud/CloudService.cpp]: can't delete this here, it's required after the read and should be cleaned up elsewhere.

File: src/Cloud/CloudService.cpp [hunk 3]
```diff
@@ -1262,6 +1261,7 @@ CloudServiceSyncDialog::syncNext()
                 QByteArray *data = new QByteArray;
                 store->readFile(data, curr->text(1), curr->text(8)); // filename
                 QApplication::processEvents();
+                delete data;
```
Comment (mtbkeith) [inline src/Cloud/CloudService.cpp]: can't delete this here, it's required after the read and should be cleaned up elsewhere.

**Pull request merged**